xchart/src/main/java/org/knowm/xchart/internal/chartpart/AxisTickCalculator_.java

```diff
@@ -74,30 +74,28 @@
       AxesChartStyler styler) {
     this.axisDirection = axisDirection;
     this.workingSpace = workingSpace;
-    Set<Double> axisValuesWithMinMax = new LinkedHashSet<>();
-    axisValuesWithMinMax.add(minValue);
-    axisValuesWithMinMax.addAll(axisValues);
-    axisValuesWithMinMax.add(maxValue);
-    this.axisValues = new ArrayList<>(axisValuesWithMinMax);
+    this.axisValues = axisValues;
     this.minValue =
         getAxisMinValue(
             styler,
             axisDirection,
             this.axisValues.stream()
                 .filter(Objects::nonNull)
+                .filter(x -> x <= minValue)
                 .mapToDouble(x -> x)
                 .min()
-                .orElseThrow(NoSuchElementException::new));
+                .orElse(minValue));
 
     this.maxValue =
         getAxisMaxValue(
             styler,
             axisDirection,
                 this.axisValues.stream()
```

Comment:
Just curious, what's the advantage of 

```
this.axisValues.stream()
                .filter(Objects::nonNull)	                .filter(Objects::nonNull)
                .filter(x -> x >= maxValue)
                .mapToDouble(x -> x)	                .mapToDouble(x -> x)
                .max()	                .max()
                .orElseThrow(NoSuchElementException::new));	                .orElse(maxValue)
```

over the original `maxValue` as I had it before?

Reply:
This was just a test that I pushed too early.